api/app/signals/apps/questionnaires/tests/services/test_answer.py

```diff
@@ -55,3 +55,42 @@ def test_validate_fieldtypes(self):
             validate_answer(123456, plaintext_question)
         with self.assertRaises(django_validation_error):
             validate_answer({'some': 'thing', 'complicated': {}}, plaintext_question)
+
+    def test_validated_selected_object(self):
+        selected_object_question = QuestionFactory(field_type='selected_object', label='selected_object',
+                                                   short_label='selected_object')
+        validate_answer = AnswerService.validate_answer_payload
+
+        # Basic payloads that should raise a validation error
+        with self.assertRaises(django_validation_error):
+            validate_answer('Invalid payload', selected_object_question)
+        with self.assertRaises(django_validation_error):
+            validate_answer(123456789, selected_object_question)
+        with self.assertRaises(django_validation_error):
+            validate_answer(True, selected_object_question)
+        with self.assertRaises(django_validation_error):
+            validate_answer(False, selected_object_question)
+
+        # These payloads should be valid
```

Comment:
This looks fine, see comments with the JSONSchema used to validate answers. I wonder how specific we should make the validation rules, this PR has quite general ones.

pro: We will not have a zoo of object selection field types (less maintenance).
con: Workflows that need to have for instance a full address will have to implement extra checks after the form has been submitted / accepted. It cannot just assume all data is present.

I think that is a subject for discussion.

Reply:
About the con you mentioned, there is no way that we can make something general that can handle everything. This implementation is the minimum needed to make sure selected objects as it is can be handled. If there are more complex cases, that can have different checks these should be implemented separately. This implementation could be basis of a different selected object implementation.